File: app/services/embedding.py
"""Embedding service wrapping OpenAI embeddings API."""
from __future__ import annotations

import asyncio
from typing import List

from openai import OpenAI

from app.core.config import settings

EMBEDDING_MODEL = "text-embedding-3-small"

_client = OpenAI(api_key=settings.openai_api_key)


async def embed_text(text: str) -> List[float]:
    """Generate an embedding vector for the given text."""

    def _embed() -> List[float]:
        response = _client.embeddings.create(model=EMBEDDING_MODEL, input=text)
        return response.data[0].embedding

    return await asyncio.to_thread(_embed)

File: services/common/src/embedding.js
import OpenAI from 'openai';
import { settings } from './config.js';

const EMBEDDING_MODEL = 'text-embedding-3-small';

const client = new OpenAI({ apiKey: settings.openAiKey });

export async function embedText(text) {
  const response = await client.embeddings.create({
    input: text,
    model: EMBEDDING_MODEL,
  });
  return {
    vector: response.data[0].embedding,
    model: EMBEDDING_MODEL,
  };
}
